stellargraph/mapper/graphwave_generator.py

```diff
@@ -0,0 +1,238 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow.keras import backend as K
+import numpy as np
+from ..core import StellarGraph
+from scipy.sparse.linalg import eigs
+from scipy.sparse import diags
+from ..core.experimental import experimental
+
+
+@experimental(
+    reason="lacks unit tests, and the time complexity could be reduced using Chebyshev polynomials.",
+    issues=[815, 853],
+)
+class GraphWaveGenerator:
+    """
+    Implementation of the GraphWave structural embedding algorithm from the paper:
+        "Learning Structural Node Embeddings via Diffusion Wavelets" (https://arxiv.org/pdf/1710.10321.pdf)
+
+    This class is minimally initialized with a StellarGraph object. Calling the flow function will return a tensorflow
+    DataSet that contains the GraphWave embeddings.
+
+    This implementation differs from the paper by removing the automatic method of calculating scales. This method was
+    found to not work well in practice, and replicating the results of the paper require manually specifying much larger
+    scales than those automatically calculated.
+    """
+
+    def __init__(self, G, scales, num_eigenvecs=-1):
```

Comment:
Oh, one more thing: why `-1` for the automatic flag? Previously you've used `"auto"` as a string, but I think a more Pythonic flag value here would be:

```suggestion
    def __init__(self, G, scales, num_eigenvecs=None):
```